### PR TITLE
ARM64: Correct the assert to add R8 as a potential valid arg register

### PR DESCRIPTION
**Pull request by briansull**
Also fix the calls to RemoveElemD to use lvVarIndex

### CODE REVIEW

**Comment by adityamandaleeka** [comment]
I assume you're aware of https://github.com/dotnet/coreclr/pull/22493/files which was also trying to address this assert?

**Comment by briansull** [comment]
@dotnet-bot test OSX10.12 x64 Checked Innerloop Build and Test
@dotnet-bot test Windows_NT arm Cross Checked Innerloop Build and Test
@dotnet-bot test Windows_NT arm64 Cross Checked Innerloop Build and Test

**Comment by jashook** [comment]
@dotnet-bot test OSX10.12 x64 Checked Innerloop Build and Test
@dotnet-bot test Windows_NT arm Cross Checked Innerloop Build and Test
@dotnet-bot test Windows_NT arm64 Cross Checked Innerloop Build and Test

**Comment by jashook** [comment]
@dotnet-bot test Windows_NT arm64 Cross Checked normal Build and Test

**Comment by jashook** [comment]
@dotnet-bot test Windows_NT arm64 Cross Checked normal Build and Test

**Comment by jashook** [comment]
@mmitche seems like we cannot request individual jenkins jobs. Is there a current issue?

@dotnet-bot test Windows_NT arm64 Cross Checked normal Build and Test

**Comment by mmitche** [comment]
@jashook Not that In know of....hmmm

**Comment by mmitche** [comment]
@dotnet-bot test Windows_NT arm64 Cross Checked normal Build and Test

**Comment by mmitche** [comment]
@dotnet-bot test OSX10.12 x64 Checked Innerloop Build and Test
@dotnet-bot test Windows_NT arm Cross Checked Innerloop Build and Test
@dotnet-bot test Windows_NT arm64 Cross Checked Innerloop Build and Test

File: src/jit/codegenarmarch.cpp [hunk 1]
```diff
@@ -2835,7 +2837,7 @@ void CodeGen::genJmpMethod(GenTree* jmp)
 
         if (compiler->lvaIsGCTracked(varDsc))
         {
-            VarSetOps::RemoveElemD(compiler, gcInfo.gcVarPtrSetCur, varNum);
+            VarSetOps::RemoveElemD(compiler, gcInfo.gcVarPtrSetCur, varDsc->lvVarIndex);
         }
 #endif
```
Comment (sandreenko) [inline src/jit/codegenarmarch.cpp]: `endif // !_TARGET_ARM64_`.

**Comment by briansull** [comment]
Blocked until #22696 goes in

**Pull request merged**